pulsar-metadata/src/test/java/org/apache/pulsar/metadata/MetadataCacheTest.java

```diff
@@ -233,7 +234,7 @@ public void insertionDeletionWitGenericType(String provider, Supplier<String> ur
         v.put("b", "2");
         objCache.create(key1, v).join();
 
-        assertEquals(objCache.getIfCached(key1), Optional.of(v));
+        assertEqualsAndRetry(() -> objCache.getIfCached(key1), Optional.of(v), Optional.empty());
```

Comment:
Can we use the Awaitibility to instread?

Reply:
We need ensure ` objCache.getIfCached(key1)` only return `Optional.empty()` or correct value.

If we use `Awaitility.await().untilAsserted(...);` here, when `objCache.getIfCached(key1)` return a wrong value it still retry.

```java
Awaitility.await().untilAsserted(() -> {
    assertEquals(objCache.getIfCached(key1), Optional.of(v));
});
```

If `Awaitility` can do what I said, please tell me. Thanks!